monitoring/db_update_sqlite.py

```diff
@@ -305,53 +304,82 @@ def refresh_BenchmarksBySubmitHost_from_view(view_name):
-                      AND UpdateTime > DATE_SUB(NOW(), INTERVAL 3 MONTH)
-                GROUP BY Site, SubmitHost, ServiceLevelType, ServiceLevel
-            ) AS latest
-            ON v.Site = latest.Site
-               AND v.SubmitHost = latest.SubmitHost
-               AND v.UpdateTime = latest.LatestPublish
-            WHERE v.EndTime > DATE_SUB(NOW(), INTERVAL 3 MONTH)
-                  AND v.UpdateTime > DATE_SUB(NOW(), INTERVAL 3 MONTH);
-        """
+                WITH latest AS (
+                    SELECT
+                        Site,
+                        SubmitHost,
+                        ServiceLevelType,
+                        ServiceLevel,
+                        UpdateTime,
+                        ROW_NUMBER() OVER (
+                            PARTITION BY Site, SubmitHost
+                            ORDER BY EndTime DESC, UpdateTime DESC
+                        ) AS rn
+                    FROM VJobRecords
+                    WHERE EndTime >= NOW() - INTERVAL 2 MONTH
+                        AND UpdateTime >= NOW() - INTERVAL 2 MONTH
+                )
+                SELECT
+                    Site,
+                    SubmitHost,
+                    ServiceLevelType,
+                    ServiceLevel,
+                    UpdateTime AS LatestPublish
+                FROM latest
+                WHERE rn = 1;
+            """
         elif view_name == 'VNormalisedSummaries':
             sql_query = f"""
-            SELECT DISTINCT v.Site, v.SubmitHost, v.ServiceLevelType, ROUND(v.NormalisedWallDuration / v.WallDuration, 3) AS ServiceLevel, v.UpdateTime AS LatestPublish
-            FROM {view_name} AS v
-            JOIN (
-                SELECT Site, SubmitHost, MAX(UpdateTime) AS LatestPublish, ROUND(NormalisedWallDuration / WallDuration, 3) AS ServiceLevel
-                FROM {view_name}
-                WHERE UpdateTime > DATE_SUB(NOW(), INTERVAL 3 MONTH)
-                    AND WallDuration > 0
-                GROUP BY Site, SubmitHost, ServiceLevelType, ServiceLevel
-            ) AS latest
-            ON v.Site = latest.Site
-               AND v.SubmitHost = latest.SubmitHost
-               AND v.UpdateTime = latest.LatestPublish
-            WHERE v.UpdateTime > DATE_SUB(NOW(), INTERVAL 3 MONTH)
-                  AND v.WallDuration > 0;
-        """
+                WITH latest AS (
+                    SELECT
+                        Site,
+                        SubmitHost,
+                        ServiceLevelType,
+                        NormalisedWallDuration,
+                        WallDuration,
+                        UpdateTime,
+                        ROW_NUMBER() OVER (
+                            PARTITION BY Site, SubmitHost
+                            ORDER BY Year DESC, Month DESC, UpdateTime DESC
+                        ) AS rn
+                    FROM VNormalisedSummaries
+                    WHERE UpdateTime >= NOW() - INTERVAL 2 MONTH
```

Comment:
swapping `DATE_SUB()` for this looks good - would it be beneficial to do this on line 159 as well (I think 159 is the only place `DATE_SUB` is used in this file now)?